source/ed/osm2ed.cpp

```diff
@@ -722,20 +716,23 @@ void PoiHouseNumberVisitor::way_callback(uint64_t osm_id, const CanalTP::Tags &t
     }
     auto logger = log4cplus::Logger::getInstance("log");
     if ((data.pois.size() + house_numbers.size()) >= max_inserts_without_bulk) {
-        persistor.insert_pois(data);
-        insert_house_numbers();
-        LOG4CPLUS_INFO(logger, "" << std::to_string(data.pois.size()) << " pois inserted and " << std::to_string(house_numbers.size()) << " house numbers");
-        n_inserted_pois += data.pois.size();
-        n_inserted_house_numbers += house_numbers.size();
-        house_numbers.clear();
-        data.pois.clear();
+        this->insert_data();
     }
 }
 
-void PoiHouseNumberVisitor::finish() {
+void PoiHouseNumberVisitor::insert_data() {
     persistor.insert_pois(data);
     persistor.insert_poi_properties(data);
     insert_house_numbers();
+    LOG4CPLUS_INFO(logger, "" << std::to_string(data.pois.size()) << " pois inserted and " << std::to_string(house_numbers.size()) << " house numbers");
```

Comment:
minor: we try to restrain line length to 113 characters, to keep readability in github